lib/ansible/modules/network/nxos/nxos_gir.py

```diff
@@ -224,8 +222,14 @@ def get_commands(module, state, mode):
             commands.append('no system mode maintenance timeout {0}'.format(
                             module.params['system_mode_maintenance_timeout']))
 
-    elif module.params['system_mode_maintenance_shutdown'] is True:
+    elif (module.params[
+        'system_mode_maintenance_shutdown'] is True and
+            mode == 'normal'):
         commands.append('system mode maintenance shutdown')
+    elif (module.params[
+          'system_mode_maintenance_shutdown'] is False and
+          mode == 'maintenance'):
```

Comment:
how about `elif not module.params['system_mode_maintenance_shutdown'] and mode == 'maintenance`?

Reply:
This cannot be done because ```system_mode_maintenance_shutdown``` is a boolean. If this parameter is completely omitted in the playbook it will be None and this condition will still be satisfied if we do the way you suggested. So I am leaving it as it.